src/worker.rs

```diff
@@ -1,14 +1,22 @@
 use nng::*;
+use serde::{Deserialize, Serialize};
 use std::collections::HashMap;
 use std::sync::atomic::Ordering;
 use std::{thread, time};
 use url::Url;
 
 use crate::goose::{GooseMethod, GooseRequest, GooseUser, GooseUserCommand};
 use crate::manager::GooseUserInitializer;
+use crate::stats::GooseTaskStats;
 use crate::util;
 use crate::{get_worker_id, GooseAttack, GooseConfiguration, WORKER_ID};
 
+#[derive(Debug, Clone, Deserialize, Serialize)]
+pub struct GaggleStats {
+    pub requests: Option<HashMap<String, GooseRequest>>,
+    pub tasks: Option<GooseTaskStats>,
+}
```

Comment:
While I like the option to make arguments optional, is there really a use-case where you want to send requests and not even an empty GooseTaskStats?

The option being implemented as enum means that you store space for both GooseTaskStats and the HashMap anyway - even if tasks was None.

Reply:
Option<> was only necessary as we could send either requests, or tasks, or both. Reworked as an enum we won't need the Option anymore.

Reply:
This now uses an Enum, defined as follow:
```rust
/// Workers send GooseMetrics to the Manager process to be aggregated together.
#[derive(Debug, Clone, Serialize, Deserialize)]
pub enum GooseMetrics {
    /// Load test hash, used to ensure all Workers are running the same load test.
    WorkerInit(u64),
    /// Goose request statistics.
    Requests(GooseRequestStats),
    /// Goose task statistics.
    Tasks(GooseTaskStats),
}
```